src/main/java/com/fasterxml/jackson/databind/JsonNode.java

```diff
@@ -608,8 +609,9 @@ public byte[] binaryValue() throws IOException {
      * types returns <code>BigInteger.ZERO</code>.
      *
      * @return {@link BigInteger} value this node contains, if numeric node; <code>BigInteger.ZERO</code> for non-number nodes.
+     * @throws StreamConstraintsException if the scale of the underlying {@link BigDecimal} is too large
      */
-    public BigInteger bigIntegerValue() { return BigInteger.ZERO; }
+    public BigInteger bigIntegerValue() throws StreamConstraintsException { return BigInteger.ZERO; }
```

Comment:
Actually.... this is backwards-incompatible change, will need to wrap exception. Since I just merged I'll change this.

Reply:
Ok, decided to use "sneaky throw" instead. Not ideal but will (have to) do for now.